File: speedtest.cpp
#include <iostream>
#include <cmath>

bool is_prime(int num) {
    // Check if a number is prime
    if (num <= 1) {
        return false;
    }
    for (int i = 2; i <= std::sqrt(num); i++) {
        if (num % i == 0) {
            return false;
        }
    }
    return true;
}

int sum_primes(int num) {
    // Calculate the sum of all prime numbers up to num
    int total = 0;
    for (int n = 0; n <= num; n++) {
        if (is_prime(n)) {
            total += n;
            // std::cout << n << " is a prime, adding it to the total. Current total: " << total << std::endl;
        }
    }
    return total;
}

int main(int argc, char* argv[]) {
    if (argc < 2) {
        std::cerr << "Please provide a number as a command-line argument." << std::endl;
        return 1;
    }

    int number = std::stoi(argv[1]);
    int sum_of_primes = sum_primes(number);
    std::cout << sum_of_primes << std::endl;

    return 0;
}

File: speedtest.py
from sys import argv

def is_prime(num: int) -> bool:
    """
    Checks if a number is prime.
    """

    if num <= 1:
        return False
    for i in range(2, int(num ** 0.5) + 1):
        if num % i == 0:
            return False
    return True

    
def sum_primes(num: int) -> int:
    """
    Calculates the sum of all prime numbers up to n.
    """

    total = 0
    
    for n in range(num + 1):
        if is_prime(n):
            total += n
            # print(f"{n} is a prime, adding it to the total. Current total: {total}")

    return total

def main():
    number = int(argv[1])

    sum_of_primes = sum_primes(number)

    print(sum_of_primes)
    
    pass

if __name__ == "__main__":
    main()
